### ISSUE TITLE
How to handle focus trap?

### ISSUE THREAD

**Issue by berenddeboer** [aforemny/elm-mdc#124]
It seems we need to support/use [the focus-trap module](https://www.npmjs.com/package/focus-trap). Else we cannot use keys in the Drawer like Esc.

How do you do this in Elm?

**Comment by berenddeboer** [comment]
Oops, should [google first](https://github.com/aforemny/elm-mdc/issues/46).

**Comment by aforemny** [comment]
Hi @berenddeboer, the focus trap is used in Elm by setting the `data-focustrap` attribute, and a MutationObserver in elm-mdc.js to monitor this attribute, which then constructs/ deconstructs the focus trap.

However, we should try not to use it unless the upstream implementation does (Dialogs). elm-mdc uses it for menus as well, but that is not great. It means you cannot have menus open from Dialogs. Do upstream Drawers use the focus trap now? Otherwise, we could look into alternatives to the problem you are trying to solve.

**Comment by berenddeboer** [comment]
Yes, drawers use focus trap, as they handle "Esc" and "key down" and "key up" now. At minimum supporting Esc would be nice.

**Comment by aforemny** [comment]
OK. In that case, just go ahead and use the focus trap. :-)

**Comment by berenddeboer** [comment]
I have something, but couldn't get it working, as when I close, it gets reopened immediately as the state is changed, and the Tick event opens the drawer immediately. I'll let you have a look at that once I have the v0.39.1 branch ready.

**Comment by aforemny** [comment]
Sounds great!
